src/pages/LoginHandler/index.js

```diff
@@ -7,140 +7,136 @@ import {
   ActionGroup,
   Button,
   Title,
+  Alert,
+  AlertActionCloseButton,
 } from '@patternfly/react-core';
 import AuthLayout from '@/components/AuthLayout';
 import { routerRedux } from 'dva/router';
 import { connect } from 'dva';
 import styles from './index.less';
 
-@connect(auth => ({
-  auth: auth.auth,
-}))
-class LoginHandler extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      username: '',
-      password: '',
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
-    };
-  }
+const mapStateToProps = state => {
+  const { auth } = state;
+  return auth;
+};
 
-  componentDidMount = () => {
-    this.disableSubmitBtn();
-  };
+const LoginHandler = props => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [btnDisabled, setBtnDisabled] = useState(true);
+  const [alertOpen, setAlertOpen] = useState(false);
+  const [alertComponent, setAlertComponent] = useState({
+    message: '',
+    variant: '',
+  });
 
-  setLoggedIn = payload => {
-    const { dispatch } = this.props;
+  const setLoggedIn = payload => {
+    const { dispatch } = props;
     dispatch({
       type: 'auth/loadUser',
       payload,
-    });
-    dispatch(routerRedux.push(`/private`));
-  };
-
-  enableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'primary',
-      btnColor: 'white',
-      btnDisabled: 'false',
-    });
-  };
-
-  disableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
+    }).then(() => {
+      dispatch(routerRedux.push(`/private`));
     });
   };
 
-  handleUserNameInputChange = username => {
-    const { password } = this.state;
-    this.setState({
-      username,
+  const handleLoginSubmit = async () => {
+    const { dispatch } = props;
+    const response = await dispatch({
+      type: 'auth/loginUser',
+      payload: {
+        username,
+        password,
+      },
     });
-    if (password !== '' && username !== '') {
-      this.enableSubmitBtn();
-    } else {
-      this.disableSubmitBtn();
-    }
-  };
-
-  handlePassWordInputChange = password => {
-    const { username } = this.state;
-    this.setState({
-      password,
+    const { status, message } = response;
+    setAlertComponent({
+      message,
+      variant: status === 'failure' ? 'danger' : 'success',
     });
-    if (username !== '' && password !== '') {
-      this.enableSubmitBtn();
-    } else {
-      this.disableSubmitBtn();
+    setAlertOpen(true);
+    if (status === 'success') {
+      // wait for a couple
+      // of seconds before taking the
+      // user to the overview page.
```

Comment:
Thanks; that makes more sense. I suggest expanding this a bit more to clarify that we're waiting to allow the user to notice the successful login alert banner before we continue.